gladier/client.py

```diff
@@ -130,6 +131,20 @@ def tools(self):
         self.__tools = [self.get_gladier_defaults_cls(gt) for gt in self.gladier_tools]
         return self.__tools
 
+    @property
+    def containers(self):
+
+        if getattr(self, '__containers', None):
+            return self.__containers
+
+        if not getattr(self, 'containers', None) or not isinstance(self.containers, Iterable):
+            raise gladier.exc.ConfigException(
+                '"containers" must be a defined list of containers paths. '
+                'Ex: ["~/.funcx/container/container.simg"]')
+        # self.__containers = [self.get_gladier_defaults_cls(c) for c in self.containers]
+        self.__containers = [c for c in self.containers]
+        return self.__containers
```

Comment:
This is interesting. How do containers work? Does the client machine need to have a local copy of a container to run it on funcx? Also, are containers intended to be run per-funcx function? 

These might be good things to bundle alongside Gladier Tools or funcx functions themselves, so they can be added automatically. Maybe this could be a value present alongside the funcx function ids in the `gladier.cfg` file that the user could override.

Reply:
Right now the containers are defined when you define a function. They also only work on a specific version of funcx. We need to talk with ZZ/Ryan in order to make this a more permanent solution.

For now. I will start adding the functionalities on the background.

Reply:
I think I need to know more about how containers work with funcX and how we want to use them. For instance, do containers just solve the dependency requirements for funcx functions? Do they also add an extra layer of security by creating an execution sandbox? If 

I don't think each Gladier Tool should need a container. But it seems like it would be good for Gladier tools to optionally provide one or allow for users to provide their own?